Source/CombatExtended/CombatExtended/Projectiles/ProjectileCE.cs

```diff
@@ -870,9 +870,20 @@ private void ApplySuppression(Pawn pawn)
             {
                 suppressionAmount = def.projectile.GetDamageAmount(1);
                 var propsCE = def.projectile as ProjectilePropertiesCE;
-                var penetrationAmount = propsCE?.armorPenetrationSharp ?? 0f;
-                var armorMod = penetrationAmount <= 0 ? 0 : 1 - Mathf.Clamp(pawn.GetStatValue(CE_StatDefOf.AverageSharpArmor) * 0.5f / penetrationAmount, 0, 1);
-                suppressionAmount *= armorMod;
+                var explodeRadius = propsCE.explosionRadius;
+                if (explodeRadius == 0f)
+                {
+                    var penetrationAmount = propsCE?.armorPenetrationSharp ?? 0f;
+                    var armorMod = penetrationAmount <= 0 ? 0 : 1 - Mathf.Clamp(pawn.GetStatValue(CE_StatDefOf.AverageSharpArmor) * 0.5f / penetrationAmount, 0, 1);
+                    suppressionAmount *= armorMod;
+                }
+                else
+                {
+                    var dPosX = ExactPosition.x - pawn.DrawPos.x;
+                    var dPosZ = ExactPosition.z - pawn.DrawPos.z;
+                    // Affected by the ratio of distance from the explosion to the max suppression radius raised to the power of two. Larger suppression amount compared to linear interpolation. Also helps that square root isn't used
+                    suppressionAmount *= Mathf.Clamp01(1f - (dPosX * dPosX + dPosZ * dPosZ) / ((explodeRadius + SuppressionRadius) * (explodeRadius + SuppressionRadius))) * explosiveSuppressionMultiplier;
```

Comment:
If you are concerned this is a hot function and want to optimize it, you should store `explodeRadius + SuppressionRadius` in a temporary local to avoid the repeated addition.  In practice, since SuppressionRadius is `const`, the JIT *should* take care of it, but worth keeping in mind.